molecule/vagrant_packager/package.py

```diff
@@ -0,0 +1,190 @@
+        """
+        return self._get_virsh_xml("dumpxml "+self.vm_name)
+
+    def default_image_location(self):
+        # type: () -> str
+        """
+            Get full system path to the default system pool dir
+        """
+        pool = self._get_virsh_xml("pool-dumpxml default")
+
+        return pool.findall('./target/path')[0].text
+
+    def image_rebase(self, img_location):
+        # type: (str) -> None
+        """ If an image has a backing store, merge the backing store into the
+            target image file
+        """
+        if self.default_image_location() in img_location:
+            raise UserWarning("To prevent catastrophy, will not"
+                              " run on image in the default pool. Copy it"
+                              " first to someplace else")
+
+        img_info = subprocess.check_output(["qemu-img", "info", img_location])
+        rebase_cmd = """qemu-img rebase -b "" {}""".format(img_location)
+
+        if "backing file:" in img_info.decode('utf-8'):
+            print("Running rebase now..")
+            subprocess.check_call(rebase_cmd, shell=True)
+        else:
+            print("No rebase necessary")
+
+    def image_store_path(self):
+        # type: () -> str
+        """ Get location of VM's first storage disk file """
+        vm_xml = self.vm_xml()
+        return vm_xml.findall('./devices/disk/source')[0].attrib['file']
+
+    def image_sparsify(self, src, dest, tmp_dir, inplace):
+        # type: (str, str, str, bool) -> None
+        """ Remove blank-space from the image. Note that setting inplace to
+            false will get you better bang for your buck but can make system
+            unstable. Pushed IO to the max on my machine and caused it to crash
+        """
+        img_info = subprocess.check_output(["qemu-img", "info", src])
+
+        if "backing file:" in img_info.decode('utf-8'):
+            raise UserWarning("Cannot sparsify image w/ backing "
+                              "store. Please rebase first.")
+
+        if inplace:
+            subprocess.check_call(["virt-sparsify",
+                                   "--in-place",
+                                   src])
+            shutil.move(src, dest)
+        else:
+            subprocess.check_call(["virt-sparsify", "--tmp",
+                                  tmp_dir,
+                                  src,
+                                  dest])
+
+    def sysprep(self, img_location):
+        # type: (str) -> None
+        """ Run the virt-sysprep tool over the image to prep the log for
+            re-dist. Removes things like logs and user history files
+        """
+        sysprep_cmd = ("virt-sysprep --no-logfile --operations "
+                       "defaults,-ssh-userdir,-ssh-hostkeys,-logfiles -a " +
+                       img_location)
+        subprocess.check_call(sysprep_cmd.split())
+
+    def vagrant_metadata(self, img_location):
+        # type: (str) -> dict
```

Comment:
This is actually wrong. We will have to install ``mypy_extensions`` and use [TypedDict](https://mypy.readthedocs.io/en/stable/kinds_of_types.html?highlight=TypedDict#typeddict) for the return type of the function.

Reply:
Good to know, @kushaldas. Currently the `typelint` check isn't running against this file. Even adding it locally to the Makefile target, I don't see any errors reported.